benches/all.rs

```diff
@@ -7,58 +10,116 @@ use primes::{
-        })
-    });
+fn bench_em(c: &mut Criterion) {
+    let mut group = c.benchmark_group("Primes");
+    // For each tuple, the first number is the largest prime number we will search for for
+    // eratosthenes-based algorithms and the second number is the index of that same prime
+    // in an array of prime numbers.
+    for i in [
+        10,
+        100,
+        1_000,
+        10_000,
+        100_000,
+        1_000_000,
+        10_000_000,
+        100_000_000,
+        500_000_000,
+    ]
+    .iter()
+    {
+        let input = Input::new(*i);
+
+        group.bench_with_input(
+            BenchmarkId::new("primal_primes", &input),
+            &input,
+            |b, input| {
+                b.iter(|| {
+                    let _: Vec<_> = PrimalPrimes::all()
+                        .take(input.largest_prime_index)
+                        .collect();
+                })
+            },
+        );
+        group.bench_with_input(
+            BenchmarkId::new("primal_sieve", &input),
+            &input,
+            |b, input| {
+                b.iter(|| {
+                    let _: Vec<_> = PrimalSieve::new(input.largest_prime as usize)
+                        .primes_from(2)
+                        .collect();
+                })
+            },
+        );
+        group.bench_with_input(BenchmarkId::new("division", &input), &input, |b, input| {
+            b.iter(|| {
+                let _: Vec<u32> = Primes::new().take(input.largest_prime_index).collect();
+            })
+        });
+        group.bench_with_input(
+            BenchmarkId::new("live_first_prime", &input),
+            &input,
+            |b, input| {
+                b.iter(|| {
+                    let _: Vec<u64> = LiveFirstPrimes::new()
+                        .take(input.largest_prime_index)
+                        .collect();
+                })
+            },
+        );
+        group.bench_with_input(
+            BenchmarkId::new("live_second_prime", &input),
+            &input,
+            |b, input| {
+                b.iter(|| {
+                    let _: Vec<u64> = LiveSecondPrimes::new()
+                        .take(input.largest_prime_index)
+                        .collect();
+                })
+            },
+        );
+        group.bench_with_input(BenchmarkId::new("sieve_og", &input), &input, |b, input| {
+            b.iter(|| {
+                let _ = og::primes(input.largest_prime);
+            })
+        });
+        group.bench_with_input(
+            BenchmarkId::new("sieve_skip_2", &input),
+            &input,
+            |b, input| {
+                b.iter(|| {
+                    let _ = skip_2::primes(input.largest_prime);
+                })
+            },
+        );
+    }
+    group.finish();
 }
 
-criterion_group!(
-    all,
-    live_first_primes,
-    live_second_primes,
-    test_division,
-    sieve_og,
-    sieve_skip_2
-);
+criterion_group!(all, bench_em,);
```

Comment:
Doesn't hurt anything, but I think the trailing comma can be removed as well, that's how this macro appears in the criterion docs.